rust/src/lib/nm/nm_dbus/nm_api.rs

```diff
@@ -271,23 +272,27 @@ impl NmApi<'_> {
         // We cannot use `org.freedesktop.NetworkManager.GetDeviceByIpIface`
         // because OVS bridge/port/iface might hold identical device name.
         // Need to check interface type also.
-        if let (Some(iface_name), Some(nm_iface_type)) =
-            (nm_conn.iface_name(), nm_conn.iface_type())
-        {
-            let nm_dev_obj_path =
-                self.get_disk_obj_path(iface_name, nm_iface_type).await?;
-            self.dbus
-                .nm_dev_reapply(nm_dev_obj_path.as_str(), nm_conn)
-                .await
-        } else {
-            Err(NmError::new(
-                ErrorKind::Bug,
-                format!(
-                    "Failed to extract interface name and type from \
-                    connection {nm_conn:?}"
-                ),
-            ))
+        if let Some(nm_uuid) = nm_conn.uuid() {
+            let nm_acs = self.active_connections_get().await?;
+            let nm_ac = nm_acs.iter().find(|ac| ac.uuid == nm_uuid);
+            if let Some(matched_ac) = nm_ac {
+                let nm_dev = nm_ac_obj_path_nm_dev_obj_path_get(
+                    &self.dbus.connection,
+                    &matched_ac.obj_path,
+                )
+                .await?;
+                if let Some(nm_dev_obj_path) = nm_dev {
+                    return self
+                        .dbus
+                        .nm_dev_reapply(nm_dev_obj_path.as_str(), nm_conn)
+                        .await;
+                }
```

Comment:
We need to raise proper error when desired connection is not activated (no NmActiveConnection found).

Reply:
Done, thank you!